src/containers/ConversionForm/index.tsx

```diff
@@ -55,21 +59,54 @@ const ConversionForm = ({
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
+    const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
+    const toggleModal = () => {
+        setFileTypeErrorModalOpen(!fileTypeErrorModalOpen);
+    };
     const toggleProcessing = () => {
         setIsProcessing(!isProcessing);
     };
 
+    const handleEngineChange = (selectedValue: string) => {
+        const selectedEngine = selectedValue as AvailableEngines;
+        setConversionEngine(selectedEngine);
+        setEngineSelected(true);
+    };
+
+    const validateFileType = () => {
+        if (fileToConvert) {
+            const fileExtension = fileToConvert.name
+                .split(".")
+                .pop()
+                ?.toLowerCase();
```

Comment:
I would prefer the reworking I suggested (it's a little lost because the code has changed) to avoid the ambiguity of what is going to be undefined here